prov/efa/src/rxr/rxr.h

```diff
@@ -728,4 +730,52 @@ static inline void rxr_rm_tx_cq_check(struct rxr_ep *ep, struct util_cq *tx_cq)
 	ofi_genlock_unlock(&tx_cq->cq_lock);
 }
 
+/*
+ * @brief: check whether we should use p2p for this transaction
+ *
+ * @param[in]	ep	rxr_ep
+ * @param[in]	efa_mr	memory registration struct
+ *
+ * @return: 0 if p2p should not be used, 1 if it should, and negative FI code
+ * if the transfer should fail.
+ */
+static inline int efa_ep_use_p2p(struct rxr_ep *rxr_ep, struct efa_mr *efa_mr)
+{
+	if (!efa_mr)
+		return 0;
+
+	/*
+	 * always send from host buffers if we have a descriptor
+	 */
+	if (efa_mr->peer.iface == FI_HMEM_SYSTEM)
+		return 1;
+
+	if (rxr_ep_domain(rxr_ep)->hmem_info[efa_mr->peer.iface].p2p_supported_by_device)
+		return (rxr_ep->hmem_p2p_opt != FI_HMEM_P2P_DISABLED);
+
+	if (rxr_ep->hmem_p2p_opt == FI_HMEM_P2P_REQUIRED) {
+		FI_WARN(&rxr_prov, FI_LOG_EP_CTRL,
+			 "Peer to peer support is currently required, but not available.\n");
+		return -FI_ENOSYS;
+	}
+
+	return 0;
+}
+
+/*
+ * @brief: check whether the endpoint supports rdma read
+ *
+ * @param[in]	domain	struct efa_domain
+ *
+ * @return: true if rdma read is supported. false otherwise.
+ */
+static inline
+bool efa_ep_support_rdma_read(struct efa_domain *domain)
```

Comment:
function name should be `efa_domain_support_rdma_read` and be moved to `efa_domain.h`

Reply:
needs `rxr_env` which is not available in `efa_domain.h

Reply:
add `rxr_env.h`  to `efa_doman.h`?

Reply:
done